unable to enter zero for rivers for fmt users

nabeelamir-defra opened the pull request:
https://eaflood.atlassian.net/browse/IWTF-1668

return error message based on if user is logged in, rather the the submission source

jaucourt reviewed (changes requested):
It's a small point, but it'd be nice to cover the new logic with tests. Would this be onerous to do?

nabeelamir-defra commented:
> It's a small point, but it'd be nice to cover the new logic with tests. Would this be onerous to do?
The thing that has changed is that instead of checking the submission to show the error message, it uses the role of the user instead. So I modified the tests to take this into account, I added WithAdminUser and WithMockUser. But I'll add some more, because there needs to be more covering the WithMockUser.

jaucourt commented:
> > It's a small point, but it'd be nice to cover the new logic with tests. Would this be onerous to do?
> 
> The thing that has changed is that instead of checking the submission to show the error message, it uses the role of the user instead. So I modified the tests to take this into account, I added WithAdminUser and WithMockUser. But I'll add some more, because there needs to be more covering the WithMockUser. Also I'll add some additional unit tests for checkMonth.

You absolutely did, I'm so sorry. This is what happens when I try to review code whilst I'm on a call! I'll approve, add those additional tests at your discretion
